csrc/quantization/cutlass_w8a8/scaled_mm_dq_c3x.cu

```diff
@@ -0,0 +1,235 @@
+#include <torch/extension.h>
+
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+// clang-format will break include orders
+// clang-format off
+#include "cutlass/cutlass.h"
+
+#include "cute/tensor.hpp"
+#include "cute/atom/mma_atom.hpp"
+#include "cutlass/numeric_types.h"
+
+#include "cutlass/gemm/device/gemm_universal_adapter.h"
+#include "cutlass/gemm/kernel/gemm_universal.hpp"
+#include "cutlass/epilogue/collective/collective_builder.hpp"
+#include "cutlass/gemm/collective/collective_builder.hpp"
+
+#include "common.hpp"
+// clang-format on
+
+using namespace cute;
+
+/////////////////////////////////////////
+
+/*
+   This defines a quantized GEMM operation with dequantized output, similar to
+   torch._scaled_mm. It is defined using the CUTLASS 3.x API, and is used for
+   NVIDIA GPUs with sm90a (Hopper) or later.
+
+   A and B may be either int8 or fp8_e4m3. A can be quantized per-tensor or
+   per-row. B can be quantized per-tensor or per-column. They must have
+   symmetric quantization.
```

Comment:
It wasn't clear to me what "symmetric quantization" refers to here -- do you mean they either both need to be per-tensor or (per-row, per-column)?

Reply:
Sorry, I'll add some more detail to clarify this comment comment. Symmetric == zero point is 0.

We can mix and match per-row and per-column. The unit tests cover all four cases

Reply:
Got it that makes sense -- I'm not as familiar with the int8 terminology unfortunately as I have mainly been working in fp8 :)